src/syntax.ml

```diff
@@ -296,10 +285,12 @@ let eq_p_config : p_config eq = fun c1 c2 ->
     are compared up to source code positions. *)
 let eq_p_command : p_command eq = fun c1 c2 ->
   match (c1.elt, c2.elt) with
-  | (P_require(m1,r1)            , P_require(m2,r2)            ) ->
-      m1 = m2 && eq_require_mode r1 r2
-  | (P_open(m1)                  , P_open(m2)                  ) ->
-      m1 = m2
+  | (P_require(ps1)              , P_require(ps2)              )
+  | (P_require_open(ps1)         , P_require_open(ps2)         )
+  | (P_open(ps1)                 , P_open(ps2)                 ) ->
+     (try List.for_all2 (=) ps1 ps2 with Invalid_argument _ -> false)
```

Comment:
This is very wrong! Polymorphic equality should not be used here. And equality should be modulo positions. Moreover, you can use `List.equal` (defined in `Extra`).

Reply:
What do you mean modulo positions? What should I put instead of (=)? Because ps1 and ps2 are of type [Path.t list].

Reply:
I was convinced that there were positions involved. If not, then there should be since we want errors to be reported for a single module, not the whole list. The argument of `P_require` should thus be of type `module_path loc`, and the equality should be `(fun mp1 mp2 -> mp1.elt = mp2.elt)`.

Reply:
Alternatively, we can just give the position of the whole statement, and print the module name for which there is a problem. This is probably simpler as a first step, since there is nothing to change. And that is probably good enough.